src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -377,7 +377,7 @@ private BoundExpression SetInferredType(BoundExpression expression, TypeSymbol t
                     {
                         var pending = (BoundDiscardExpression)expression;
                         Debug.Assert((object?)pending.Type == null);
-                        return pending.SetInferredTypeWithAnnotations(TypeWithAnnotations.Create(type));
+                        return pending.SetInferredTypeWithAnnotations(TypeWithAnnotations.Create(true, type, true));
```

Comment:
It wasn't clear to me why we don't just want to call `TypeWithAnnotations.Create(type)` here. An oblivious annotation seems fine here, just as we have on `DeconstructionVariablePendingInference` above.